### PR TITLE
Update API docs with defaulting and validation

### PR DESCRIPTION
**Pull request by alculquicondor**
#### What type of PR is this?

/kind documentation

#### What this PR does / why we need it:

Update API docs to reflect changes in webhooks.

#### Which issue(s) this PR fixes:

Fixes #

### CODE REVIEW

File: apis/kueue/v1alpha1/workload_types.go [hunk 1]
```diff
@@ -76,6 +81,9 @@ type PodSet struct {
 	Name string `json:"name"`
 
 	// spec is the Pod spec.
+	// If requests are omitted for a container or initContainer,
+	// they defaults to the limits, if they are explicitly specified for the
```
Comment (ahg-g) [inline apis/kueue/v1alpha1/workload_types.go:85]:
```suggestion
	// they default to the limits if they are explicitly specified for the
```

Comment (alculquicondor) [reply apis/kueue/v1alpha1/workload_types.go]: Done

**Pull request merged**